File: signature/momentjs.go
package signature

var MomentJsSignature = Signature{
	Name:        "Moment.js",
	Description: "Parse, validate, manipulate, and display dates and times in JavaScript.",
	Cpe:         "cpe:/a:jqueryui:jquery_ui",

	DetectPattern: DetectPattern{
		URLs: []string{
			"moment",
		},
		Bodies: []string{
			"moment",
		},
	},
	VersionPattern: VersionPattern{
		Bodies: []string{
			`//! version : (\d+\.\d+\.\d+)`,
			`VERSION = '(\d+\.\d+\.\d+)'`,
			`b.version="(\d+\.\d+\.\d+)"`,
		},
	},
}

File: src/signatures/technologies/moment_js.ts
import type { Signature } from "../_types.js";

export const momentJsSignature: Signature = {
  name: "Moment.js",
  description:
    "Moment.js is a free and open-source JavaScript library that removes the need to use the native JavaScript Date object directly.",
  cpe: "cpe:2.3:a:momentjs:moment:*:*:*:*:*:*:*:*",
  rule: {
    confidence: "high",
    urls: ["moment(?:\\.min)?\\.js"],
    javascriptVariables: {
      moment: "",
      "moment.version": "^(.+)$",
    },
  },
};
